src/Transpiler.ts

```diff
@@ -805,6 +805,11 @@ export abstract class LuaTranspiler {
                 return this.transpileSpreadElement(node as ts.SpreadElement);
             case ts.SyntaxKind.NonNullExpression:
                 return this.transpileExpression((node as ts.NonNullExpression).expression);
+            case ts.SyntaxKind.ClassExpression:
+                this.namespace.push("");
```

Comment:
Any specific reason for this line?

Reply:
It scopes the class declaration. Otherwise the class would be in global scope. Not sure if this is the best approach - I see it as sort of an anonymous namespace in C++.

Reply:
Could you provide an input/ouput example?

Reply:
`let f = function()
{
    let a = class{}
}`
would generate 
`local f; f = function()
    local a = (function()   --[[ should be local ]]  _ = _ or {}
    _.__index = _
    function _.new(construct, ...)
        local self = setmetatable({}, _)
        if construct and _.constructor then _.constructor(self, ...) end
        return self
    end
    function _.constructor(self)
    end
; return _ end)();
end
;`